src/Akka.Hosting.TestKit/TestKit.cs

```diff
@@ -102,19 +102,6 @@ private void InternalConfigureServices(HostBuilderContext context, IServiceColle
                     });
                 }
 
-                builder.StartActors((system, registry) =>
-                {
-                    try
-                    {
-                        base.InitializeTest(system, (ActorSystemSetup)null!, null, null);
```

Comment:
I think this was part of our problem - the synchronization context we use to keep track of the implicit `TestActor` was being created, possibly but not always, on a separate thread from the main test execution context.